sys/vm/vm_map.c

```diff
@@ -4363,10 +4363,10 @@ vm_map_delete(vm_map_t map, vm_offset_t start, vm_offset_t end,
 	 * Find the start of the region, and clip it.
 	 * Step through all entries in this region.
 	 */
-	rv = vm_map_lookup_clip_start(map, start, &entry, &scratch_entry);
+	rv = vm_map_lookup_clip_start(map, start, &entry, &prev_entry);
 	if (rv != KERN_SUCCESS)
 		return (rv);
-	for (; entry->start < end; entry = next_entry) {
+	for (; entry->start < end; prev_entry = entry, entry = next_entry) {
```

Comment:
So this is the real fix, updating prev_entry when iterating the loop?

Reply:
Yeah, updating `prev_entry` is the fix. The rest of renaming it from `scratch_entry` to `prev_entry`.